src/py42/modules/securitydata.py

```diff
@@ -169,6 +176,66 @@ def search_file_events(self, query):
         file_event_client = self._microservices_client_factory.get_file_event_client()
         return file_event_client.search(query)
 
+    def _search_by_hash(self, hash, type):
+        query = FileEventQuery.all(type.eq(hash))
+        response = self.search_file_events(query)
+        return response["fileEvents"]
+
+    def _find_file_versions(self, md5_hash, sha256_hash):
+        file_event_client = self._microservices_client_factory.get_file_event_client()
+        pds_client = self._microservices_client_factory.get_preservation_data_service_client()
+        response = file_event_client.get_file_location_detail_by_sha256(sha256_hash)
+        for device_id, paths in parse_file_location_response(response):
```

Comment:
the inside of this for loop should should have a `try/except` block (where the except simply passes) so that if we hit an error when trying to find file versions for a specific file id/path, we will keep trying the others in the list.

Reply:
I understand the intent behind why we wanted to do that. However, we would be hiding errors/api-errors or incorrect client request errors in doing so. If in future the api introduces a bug or breaking change we would just end up ignoring them.

Allowing errors to pass would end up having similar effect as drawbacks of catching generic exception.

If there are particular cases we foresee that these APIs return which indicate kind of "a success state but cannot fulfil the request now ", we should be catching those errors and allowing them to pass.

As in below, I would like to change line 193 as :
      `if response.status_code == 204:
           continue`
since that is a special case where the state is success but there is no file to download.

Reply:
> Allowing errors to pass would end up having similar effect as drawbacks of catching generic exception.
Not quite. We would only be allowing errors that we _expect_ could happen to pass. If there was any type of error besides `Py42HttpError`, we would still raise.

I understand your point about hiding api errors, we should be telling users this information -- so my suggestion would be that if a `Py42HttpError` occurs, we should log it with `py42.settings.debug.logger.warn()` (but still not throw it). We could include an advisement to turn on debug logging in the top-level error that occurs if none of the entries work.

Reply:
Yes, makes sense, I will add the logging.

Reply:
Done.